crud/compare/keydef.lua

```diff
@@ -55,6 +58,10 @@ local function new(space, field_names, index_id)
         keydef = keydef:merge(keydef_lib.new(normalize_parts(updated_parts)))
     end
 
+    if keydef_lib.__is_external and not merger_lib.__is_external then
```

Comment:
Maybe it's better to save result to `need_cast` variable to check only one flag?

Reply:
I don't think that it makes sense, these modules are required separately, `__is_external ` is set for each of them. This condition is checked only once here.

Reply:
> This condition is checked only once here.

This condition is checked for each key_def.new

Reply:
Why don't cast always (and add a comment that would describe the situation)? It would simplify logic.

Reply:
See https://github.com/tarantool/crud/actions/runs/957320219. It fails for 1.10 with `'struct key_def' has no member named 'compare'` error.

Reply:
Sometimes it shoots.

![](https://user-images.githubusercontent.com/8830475/122798417-3d155600-d2c9-11eb-8f2a-f4fbc491531d.jpg)

On 1.10 when merger is external.

Reply:
Ah, it seems, got it. When there is no built-in key_def module, we have no metatype with __index.compare for `cdata<struct key_def &>`. But crud calls this method somewhere.

I guess there should not be a problem, if we'll cast right before passing key_def to merger. It also better, because we'll not loss the link to the original key_def, so it'll not garbage collected.

Reply:
Thanks for your idea!